src/App.tsx

```diff
@@ -1,8 +1,141 @@
 import React from 'react';
 import './App.css';
+import { AppState, Todo, ControlPanel } from './components/interfaces';
+import { DownloadButton } from './components/DownloadButton';
+import { allTodosDatas } from './components/getTodos';
+import ListTodos from './components/TodosList';
+import { SortedPanel } from './components/SortedPanel';
 
-const App = () => (
-  <h1>Dynamic list of TODOs</h1>
-);
+class App extends React.Component {
+  state: AppState = {
+    todos: [],
+    order: {
+      sortId: true,
+      sortName: false,
+      sortTitle: false,
+      sortStatus: false,
+    },
+    isLoading: false,
+    isLoadDatas: false,
+  };
+
+  getTodos = () => {
+    this.setState({ isLoading: true });
+    allTodosDatas()
+      .then(todos => this.setState({ todos, isLoadDatas: true }))
+      .finally(() => this.setState({ isLoading: false }));
+  };
+
+  sortByName = () => {
+    const { todos } = this.state;
+    const { sortName } = this.state.order;
+    let sortedTodos: Todo[];
+
+    if (!sortName) {
+      sortedTodos = [...todos
+        .sort((a: Todo, b: Todo) => a.user?.username.localeCompare(b.user?.username)),
+      ];
+    } else {
+      sortedTodos = [...todos
+        .sort((a: Todo, b: Todo) => b.user?.username.localeCompare(a.user?.username)),
+      ];
+    }
+
+    const changeOrder = !sortName;
+
+    this.setState(({ todos: sortedTodos, order: { sortName: changeOrder } }));
+  };
+
+  sortByTitle = () => {
+    const { todos } = this.state;
+    const { sortTitle } = this.state.order;
+    let sortedTodos: Todo[];
+
+    if (!sortTitle) {
+      sortedTodos = [...todos
+        .sort((a: Todo, b: Todo) => a.title.localeCompare(b.title)),
+      ];
+    } else {
+      sortedTodos = [...todos
+        .sort((a: Todo, b: Todo) => b.title.localeCompare(a.title)),
+      ];
+    }
+
+    const changeOrder = !sortTitle;
+
+    this.setState(({ todos: sortedTodos, order: { sortTitle: changeOrder } }));
+  };
+
+  sortByStatus = () => {
+    const { todos } = this.state;
+    const { sortStatus } = this.state.order;
+    let sortedTodos: Todo[];
+    const completedTodos: Todo[] = todos.filter(todo => todo.completed);
+    const activeTodos: Todo[] = todos.filter(todo => !todo.completed);
```

Comment:
todos.sort(a.completed - b.completed) or todos.sort(b.completed - a.completed)
you don't need two separate arrays